integrationtest/mapper/pojo-base/src/test/java/org/hibernate/search/integrationtest/mapper/pojo/mapping/definition/IndexedEmbeddedBaseIT.java

```diff
@@ -1617,10 +1822,682 @@ class EntityC {
 								"Path starting from that type and ending with a cycle: 'c.b.c.'",
 								"A type cannot declare an unrestricted @IndexedEmbedded to itself, even indirectly",
-								"To break the cycle, you should consider adding filters to your @IndexedEmbedded: includePaths, includeDepth, ..." )
+								"To break the cycle, you should consider adding filters to your @IndexedEmbedded: includePaths, includeDepth, excludePaths, ..." )
+				);
+	}
+
+	/*
+	 * We are trying to include something that is not reachable
+	 * (even though the property is there it is explicitly excluded by a child, so we must fail)
+	 */
+	@Test
+	public void parentIncludeChildExclude() {
+		class Model {
+			@Indexed(index = INDEX_NAME)
+			class IndexedEntity {
+				@DocumentId
+				Integer id;
+				@KeywordField
+				String indexedEntityString;
+
+				@IndexedEmbedded(includePaths = { "includedString", "subIncluded.subSubIncluded.subSubIncludedString" })
+				Included included;
+
+				public IndexedEntity(Integer id, String indexedEntityString, Included included) {
+					this.id = id;
+					this.indexedEntityString = indexedEntityString;
+					this.included = included;
+				}
+			}
+
+			class Included {
+				@KeywordField
+				String includedString;
+
+				@IndexedEmbedded
+				SubIncluded subIncluded;
+
+				public Included(String includedString, SubIncluded subIncluded) {
+					this.includedString = includedString;
+					this.subIncluded = subIncluded;
+				}
+			}
+
+			class SubIncluded {
+				@KeywordField
+				String subIncludedString;
+
+				@IndexedEmbedded(excludePaths = "subSubIncludedString")
+				SubSubIncluded subSubIncluded;
+
+				public SubIncluded(String subIncludedString) {
+					this.subIncludedString = subIncludedString;
+				}
+			}
+
+			class SubSubIncluded {
+				@KeywordField
+				String subSubIncludedString;
+
+				public SubSubIncluded(String subSubIncludedString) {
+					this.subSubIncludedString = subSubIncludedString;
+				}
+			}
+		}
+
+		assertThatThrownBy( () -> setupHelper.start()
+				.withAnnotatedEntityTypes( Model.IndexedEntity.class )
+				.setup() )
+				.isInstanceOf( SearchException.class )
+				.satisfies( FailureReportUtils.hasFailureReport()
+						.typeContext( Model.IndexedEntity.class.getName() )
+						.failure(
+								"An @IndexedEmbedded defines includePaths filters that do not match anything.",
+								"Non-matching includePaths filters: [subIncluded.subSubIncluded.subSubIncludedString].",
+								"Encountered field paths: [includedString, subIncluded, subIncluded.subIncludedString]. Check the filters for typos, or remove them if they are not useful."
+						)
+				);
+	}
+
+	/*
+	 * Child has a property (that we included explicitly), so if we are trying to exclude it at a parent level it should
+	 * get excluded without complains. *BUT*, because it's a single field in that embedded we end up dropping the entire embedded
+	 * resulting in an error.
```

Comment:
Okay so this is confusing... I'd expect no error in this case, or at least not something that tells me `subIncluded.subSubIncluded.subSubIncludedString` doesn't exist, since it actually does.

> because it's a single field in that embedded we end up dropping the entire embedded

This is fine, but since the dropping was triggered by the exclusion of `subIncluded.subSubIncluded.subSubIncludedString`, I'd expect that exclusion to have marked the path as encountered, and then we would not report the filter as useless... ? Isn't something wrong here?

Reply:
+1 On it being not very clear. One could try to make an argument that `subIncluded.subSubIncluded.subSubIncludedString` is a useless filter as in such case `subIncluded.subSubIncluded` would do the same. 
The particular problem comes from the fact that we eliminate the entire embedded by `IndexSchemaFilter#isEveryPathExcluded()` which explicitly asks not to track paths... Let me think if we can catch that in some clean way ...